UKHomeOffice/nmsw-ui
# NMSW-33 Add basic autocomplete component

# Ticket

NMSW-33

----

## AC

Given I have to select from a predefined list of items
When I start to type an items name (After 2 characters)
Then I am shown a list of options that could match that item (From start of word)

----

## To test

- run app locally
- go to second page form
- go to country question
- type 2 letters of a country
- > you should see countries that match those letters in the list
- select a country
- go to the port question
- type `do` (we only have dover and portsmouth ports in the list right now)
- > you should see options for `Dover GB DOV` and `Dover Marina` and `Dover Other GB POR`
- select a port
- fill out rest of form
- click submit
- form should submit
- confirmation page reference should include the country and port you selected

----

## Notes

- there is no persist data on refresh yet
- there is no error handling yet
- we do not yet pass through country code or port code as separate fields in the dataset
